libs/cardano-ledger-test/src/Test/Cardano/Ledger/Constrained/Preds/CertState.hs

```diff
@@ -111,12 +145,12 @@ pstateGenPreds _ =
       , (1, Constr "(+3)" (+ 3) ^$ e, [CanFollow e currentEpoch])
       , (1, Constr "(+5)" (+ 5) ^$ e, [CanFollow e currentEpoch])
       ]
-      -- poolDistr not needed in PState, but is needed in NewEpochState
-      -- But since it is so intimately tied to regPools we define it here
-      -- Alternately we could put this in NewEpochState, and insist that pStateStage
-      -- preceed newEpochStateStage
-      -- , Dom regPools :=: Dom poolDistr
-      -- , SumsTo (Right (1 % 1000)) (Lit RationalR 1) EQL [ProjMap RationalR individualPoolStakeL poolDistr]
+  , -- poolDistr not needed in PState, but is needed in NewEpochState
+    -- But since it is so intimately tied to regPools we define it here
+    -- Alternately we could put this in NewEpochState, and insist that pStateStage
+    -- preceed newEpochStateStage
+    Dom regPools :=: Dom poolDistr
```

Comment:
I would be very happy if we could move these constraints to where they are needed instead. Having variables used in constraints that don't relate to that type can cause some real headaches in properties.

Reply:
So the best I could do was move them to the Ledger stage, as it needs poolDistr to construct a valid pulser.